clp_ffi_py/ir/native_deprecated.py

```diff
@@ -0,0 +1,138 @@
+    " :class:`~clp_ffi_py.ir.native.FourByteSerializer` instead.",
+)
+class FourByteEncoder:
+    @staticmethod
+    def encode_preamble(ref_timestamp: int, timestamp_format: str, timezone: str) -> bytearray:
+        """
+        This method is deprecated.
+
+        Please check
+        :meth:`~clp_ffi_py.ir.native.FourByteSerializer.serialize_preamble`.
+        """
+        return FourByteSerializer.serialize_preamble(ref_timestamp, timestamp_format, timezone)
+
+    @staticmethod
+    def encode_message_and_timestamp_delta(timestamp_delta: int, msg: bytes) -> bytearray:
+        """
+        This method is deprecated.
+
+        Please check
+        :meth:`~clp_ffi_py.ir.native.FourByteSerializer.serialize_message_and_timestamp_delta`.
+        """
+        return FourByteSerializer.serialize_message_and_timestamp_delta(
+            timestamp_delta,
+            msg,
+        )
+
+    @staticmethod
+    def encode_message(msg: bytes) -> bytearray:
+        """
+        This method is deprecated.
+
+        Please check
+        :meth:`~clp_ffi_py.ir.native.FourByteSerializer.serialize_message`.
+        """
+        return FourByteSerializer.serialize_message(msg)
+
+    @staticmethod
+    def encode_timestamp_delta(timestamp_delta: int) -> bytearray:
+        """
+        This method is deprecated.
+
+        Please check
+        :meth:`~clp_ffi_py.ir.native.FourByteSerializer.serialize_timestamp_delta`.
+        """
+        return FourByteSerializer.serialize_timestamp_delta(timestamp_delta)
+
+    @staticmethod
+    def encode_end_of_ir() -> bytearray:
+        """
+        This method is deprecated.
+
+        Please check
+        :meth:`~clp_ffi_py.ir.native.FourByteSerializer.serialize_end_of_ir`.
+        """
+        return FourByteSerializer.serialize_end_of_ir()
+
+
+@deprecated(
+    version="0.0.13",
+    reason=":class:`DecoderBuffer` is deprecated; use"
+    " :class:`~clp_ffi_py.ir.native.DeserializerBuffer` instead.",
+)
+class DecoderBuffer:
+    def __init__(self, input_stream: IO[bytes], initial_buffer_capacity: int = 4096):
+        self._deserializer_buffer = DeserializerBuffer(
+            input_stream=input_stream, initial_buffer_capacity=initial_buffer_capacity
+        )
+
+    def get_num_decoded_log_messages(self) -> int:
+        """
+        Please check
+        :meth:`~clp_ffi_py.ir.native.DeserializerBuffer.get_num_deserialized_log_messages`.
+        """
+        return self._deserializer_buffer.get_num_deserialized_log_messages()
+
+    def _test_streaming(self, seed: int) -> bytearray:
+        """
+        Please check :meth:`~clp_ffi_py.ir.native.DeserializerBuffer._test_streaming`.
+        """
+        return self._deserializer_buffer._test_streaming(seed)
+
+
+@deprecated(
+    version="0.0.13",
+    reason=":class:`Decoder` is deprecated; use"
+    " :class:`~clp_ffi_py.ir.native.FourByteDeserializer` with"
+    " :class:`~clp_ffi_py.ir.native.DeserializerBuffer` instead.",
```

Comment:
```suggestion
    reason=":class:`Decoder` is deprecated and has been renamed to"
    " :class:`~clp_ffi_py.ir.native.FourByteDeserializer`.",
```

I think they'll get the warning about DecoderBuffer already, right? If so, I don't think we need to add more detail here.